test/widgets/subscription_list_test.dart

```diff
@@ -190,4 +202,47 @@ void main() {
     check(tester.widget<UnreadCountBadge>(find.byType(UnreadCountBadge)).backgroundColor)
       .equals(swatch);
   });
+
+  testWidgets('muted streams are displayed as faded', (tester) async {
+
+    testOpacityForStreamAndBadge(String streamName, double opacity) {
+      final streamFinder = find.text(streamName);
+      final streamOpacity = tester.widget<Opacity>(
+        find.ancestor(of: streamFinder, matching: find.byType(Opacity))).opacity;
+      final badgeOpacity = tester.widget<Opacity>(find.ancestor(of: find.descendant(
+        of: find.ancestor(of: streamFinder, matching: find.byType(Row)),
+        matching: find.byType(UnreadCountBadge)), matching: find.byType(Opacity))).opacity;
+      check(streamOpacity).equals(opacity);
+      check(badgeOpacity).equals(opacity);
+    }
+
+    final stream1 = eg.stream(name: 'Stream 1');
+    final stream2 = eg.stream(name: 'Stream 2');
+    final unreadMsgs = eg.unreadMsgs(streams: [
+      UnreadStreamSnapshot(streamId: stream1.streamId, topic: 'a', unreadMessageIds: [1, 2]),
+      UnreadStreamSnapshot(streamId: stream2.streamId, topic: 'b', unreadMessageIds: [3]),
+    ]);
+    await setupStreamListPage(tester,
+      subscriptions: [
+        eg.subscription(stream1, isMuted: true),
+        eg.subscription(stream2, isMuted: false)
+      ],
+      userTopics: [
+        UserTopicItem(
+          streamId: stream1.streamId,
+          topicName: 'a',
+          lastUpdated: 1234567890,
+          visibilityPolicy: UserTopicVisibilityPolicy.unmuted,
```

Comment:
This can be made a bit more compact (and with less noise from boring parts like `lastUpdated`) with an appropriate helper.

There's one already in `model/stream_test.dart`. So that can be moved to `example_data.dart` in a prep commit.

Reply:
bump — you added that useful prep commit but then didn't actually use the new shared helper here :slightly_smiling_face: